README.md

------------------------------------------------------------------------

------------------------------------------------------------------------

[![Project Status: WIP ? Initial development is in progress, but there has not yet been a stable, usable release suitable for the public.](http://www.repostatus.org/badges/latest/wip.svg)](http://www.repostatus.org/#wip)

------------------------------------------------------------------------

[![minimal R version](https://img.shields.io/badge/R%3E%3D-NA-6666ff.svg)](https://cran.r-project.org/) [![CRAN\_Status\_Badge](http://www.r-pkg.org/badges/version/spanishRshapes)](https://cran.r-project.org/package=spanishRshapes) [![packageversion](https://img.shields.io/badge/Package%20version-0.1.0-orange.svg?style=flat-square)](commits/master)

------------------------------------------------------------------------

[![Last-changedate](https://img.shields.io/badge/last%20change-2018--02--06-yellowgreen.svg)](/commits/master)

<!-- README.md is generated from README.Rmd. Please edit that file -->
spanishRshapes <img src="man/figures/imgfile2.png" align="right" />
===================================================================

The goal of spanishRshapes es facilitar el uso de datos de limites territoriales españoles en R. Contiene ficheros de datos para los lindes de: CC.AA. provincias y municipios. Los datos proviene del del [Instituto Geográfico Nacional](http://www.ign.es/web/ign/portal), concretamente del [centro de descargas](http://centrodedescargas.cnig.es/CentroDescargas/index.jsp). Una vez en el centro de descargas, hay que ir a `Información geográfica de referencia` y luego a `Lineas límite municipales`

Conuntos de datos
=================

`spanishRshapes` es un package de datos. Concretamente contiene los siguientes conjuntos de datos:

-   **IGN\_CCAA\_17**: lindes/polígonos/shapes de las 17 CC.AA españolas (+ Ceuta y Melilla). \[19 rows y 11 variables\]

-   **IGN\_prov\_17**: lindes/polígonos/shapes de las provincias españolas (52 rows and 13 variables)

-   **IGN\_mun\_17**: lindes/polígonos/shapes de los municipios españoles (8211 rows and 20 variables). \[8.124 municipios, 81 condominios, 6 islitas alrededor de la Gomera\]

Formato de datos
================

Los datos estan en spatial-df.

Uso
===

Para cargar los lindes provinciales:

``` r
devtools::install_github("perezp44/spanishRshapes")
library(spanishRshapes)
lindes_prov <- IGN_prov_17
```


NEWS.md
# spanishRshapes 0.1.0

* Added a `NEWS.md` file to track changes to the package.


_script.R
#1) En RStudio: File >> New Project >>New Directory >> R pkg (sin poner git)
#2) Borro ./R/hello.R y ./man/hello.Rd


library(usethis)
#---------------- has de elegir un nombre para el pkg
name_of_the_pkg <- "spanishRshapes"
#path <- paste0("C:/Users/perezp/Desktop/a_GIT_2016a/", name_of_the_pkg)
#create_package(path, rstudio = TRUE)

#------------------------
use_roxygen_md()  #- sets up roxygen2 and enables markdown mode so you can use markdown in your roxygen2 comment blocks.
use_package_doc() #- creates un archivo en ./R/  es a skeleton documentation file for the complete package, taking the advantage of the latest roxygen2 features to minimise duplication between the DESCRIPTION and the documentation
use_readme_rmd()  #- creates a README.Rmd: use this to describe what your package does and why people should care about it.
use_news_md()     #- creates a basic NEWS.md for you to record changes
use_mit_license(name = "Pedro J. Pérez")

#----------------------- DESCRIPTION
#- Modifico las siguientes lineas:
# Title: Provides some Spanish Boundary data (shapes) as 'sf'-Ready Data Frames
# Description:  Provides Spanish Boundary data (shapes) as 'sf'-Ready Data Frames. Boundary Data comes from
# Authors@R: person("Pedro J.", "Perez", , "pedro.j.perez@uv.es", c("aut", "cre"))
# Maintainer: Pedro J. Perez <pjperez@uv.es>


#----------------  Vignette
my_name_vignette <- paste0("intro-to-", name_of_the_pkg)
use_vignette(my_name_vignette) #- sets you up for success by configuring DESCRIPTION and creating a .Rmd template in vignettes/
#- Has de poner en el yaml de la vignette
# title: "Intro to spanishRshapes package"
# author: "Pedro J. Pérez"
# %\VignetteIndexEntry{Intro to spanishRshapes package}


#----------------------
devtools::use_data_raw()  #- crea directorio de raw-data

#----------------------
use_build_ignore("_script.R")





#--------------------------- GIT
use_git(message = "Primer commit")           #- activa GIT. se reinicia RStudio, asi q has de volver a hacer library(usethis)



#---------------------------- GITHUB
library(usethis)
use_github(protocol = "https") #- AQUI sale un error
use_github_labels()  #- labelling issues
use_github_links()   #- añade links en la file DESCRIPTION



#--------------------- te vas a TERMINAL y haces
# git add -A
# git commit --all --message "todo a Github"
# git push -u origin master
#

#--------------------------- Ahora ya toca meter datos y arreglar documentación
#--------------------------- Ahora ya toca meter datos y arreglar documentación


#--------------------  README
#------ has de knittear README.Rmd para que aparezca README.md q es la que se muestra en Github por defecto


#--------------------- LA VIGNETTE
use_build_ignore("./vignettes/info_vignettes.txt")




#------------------------------------------------------
#------------------------------------------------------ PARA DESPUES CORRERLO
#------------------------------------------------------


devtools::build_vignettes()  #- viñetas

devtools::check()            #- chequea


devtools::document()   #- procesa los roxygen comments y las vignettes



devtools::build()



#------------------------------ USAR el pkg



# devtools::install_github("perezp44/spanishRshapes")
library(spanishRshapes)
library(tidyverse)

aa <- ls("package:spanishRshapes", all = TRUE) %>% as.data.frame()#- ves lo que hay en mypkgDataforblog
help(package = spanishRshapes)
bb <- IGN_mun_17s
names(bb)
cc <- bb %>% filter(NombreMuni  == "Pancrudo")
bb[bb$NombreCCAA=="Galicia",]

#--------------------   Github
# git remote add origin https://github.com/perezp44/mypkgfordata.git
# cd c:/Users/perezp/Desktop/a_GIT_2016a/mypkgDataforblog
# git add -A    # stages all files
# git commit --all --message "Creando el REPO"
# git push -u origin master


#----------------- Licencia
# En esta pagina pone esto de licencia
# http://opendata.esri.es/datasets/d8854f26fd5c4baab08337ca0f3aff6f_0#
#Licencia (compatible con CC-BY 4.0) ampara el uso libre y gratuito para cualquier propósito legítimo,
#siendo la única estricta obligación la de reconocer y mencionar el origen y propiedad de los productos
#y servicios de información geográfica licenciados como del IGN según se indica en la licencia.
#Créditos: © Instituto Geográfico Nacional



#------- BADGES
install.packages("badgecreatr")
badgecreatr::badgeplacer( githubaccount = "perezp44",githubrepo = "spanishRshapes", branch = "master")
badgecreatr::badgeplacer(location = ".", status = "wip" , githubaccount = "perezp44",githubrepo = "spanishRshapes", branch = "master")




#------- PKGDOWN


R/data_INE.R
#------------------------------ This file documents INE datasets

#' Codigos INE para provincias y CC.AA. (SPAIN)
#'
#' Codigos y nombres oficiales de las provincias y CCAA (SPAIN). Sacados del INE.
#' 52 provincias (51 = Ceuta y 52 = Melilla).
#' 19 CC.AA (18 = Ceuta y 19 = Melilla).
#'
#' @source \url{http://www.ine.es}
#'         \url{http://www.ine.es/dyngs/INEbase/es/operacion.htm?c=Estadistica_C&cid=1254736177031&menu=ultiDatos&idp=1254734710990}
#'
#' @format A data frame with 52 rows and 4 variables:
#' \itemize{
#'   \item CODAUTO: Código de la C.A.
#'   \item CCAA: Nombre de la C.A.
#'   \item CPRO: Código de la provincia.
#'   \item PROVINCIA: Nombre de la provincia.
#' }
#'
#' @examples
#' \dontrun{
#'  df <- cod_INE_prov
#' }
#'
"cod_INE_prov"



#' Codigos INE municipios 2017
#'
#' Codigos del INE para municipios españoles a 1 enero 2017: 8.124 municipios, 6 variables.
#'
#' @source \url{http://www.ine.es/daco/daco42/codmun/codmunmapa.htm}
#'
#' @format A data frame with 8124 rows and 4 variables: Cerdedo-Cotobade se unieron en 2016
#' \itemize{
#'   \item CODAUTO: Código de CCAA
#'   \item CPRO: Código de la provincia
#'   \item CMUN: Código del municipio
#'   \item DC: Código de control
#'   \item NOMBRE: Nombre del municipio
#'   \item INECodMuni: CPRO+CMUN (5 digits code)
#' }
#'
"cod_INE_muni_17"


#' Codigos INE municipios 2016
#'
#' Codigos del INE para municipios españoles a 1 enero 2016: 8.125 municipios, 6 variables.
#' Un municipio más que en 2017
#'
#' @format A data frame with 8125 rows (1 municipio mas que en 2017) and 6 variables :
#' \itemize{
#'   \item CODAUTO: Código de CCAA
#'   \item CPRO: Código de la provincia
#'   \item CMUN: Código del municipio
#'   \item DC: Código de control
#'   \item NOMBRE: Nombre del municipio
#'   \item INECodMuni: CPRO+CMUN (5 digits code)
#' }
#'
#' @source \url{http://www.ine.es}
#'
"cod_INE_muni_16"




#' Codigos INE municipios 2017 (1 enero)
#'
#' Codigos del INE para municipios españoles a 1 enero 2017 (tb tiene los codigos de provincia y CCAA)
#'
#' @format A data frame with 8124 rows and 11 variables:
#' \itemize{
#'   \item INECodCCAA: Código de CCAA
#'   \item INECodProv: Código de la provincia
#'   \item CMUN: Código del municipio (es del INE, pero sin el codigo de la Provincia no es útil)
#'   \item DC: Código de control
#'   \item NombreMuni: Nombre del municipio
#'   \item INECodMuni: Código de provincia + Código municipio. Es el que usa el INE muchas veces (5 digitos)
#'   \item NombreCCAA: Nombre de la CCAA
#'   \item NombreProv: Nombre de la provincia
#'   \item INENumMuni: número del municipio (as.integer(INECodMuni))
#'   \item INENumProv: número de la provincia (as.integer(INECodProv))
#'   \item INENumCCAA: número de la CCAA (as.integer(INECodCCAA))
#' }
#'
#'
#' @source \url{http://www.ine.es}
"cod_muni_pjp_17"



#' Codigos INE municipios 2016 (1 enero)
#'
#' Codigos del INE para municipios españoles a 1 enero 2016 (tb tiene los codigos de provincia y CCAA)
#'
#' @format A data frame with 8125 rows and 11 variables:
#' \itemize{
#'   \item INECodCCAA: Código de CCAA
#'   \item INECodProv: Código de la provincia
#'   \item CMUN: Código del municipio (es del INE, pero sin el codigo de la Provincia no es útil)
#'   \item DC: Código de control
#'   \item NombreMuni: Nombre del municipio
#'   \item INECodMuni: Código de provincia + Código municipio. Es el que usa el INE muchas veces (5 digitos)
#'   \item NombreCCAA: Nombre de la CCAA
#'   \item NombreProv: Nombre de la provincia
#'   \item INENumMuni: número del municipio (as.integer(INECodMuni))
#'   \item INENumProv: número de la provincia (as.integer(INECodProv))
#'   \item INENumCCAA: número de la CCAA (as.integer(INECodCCAA))
#' }
#'
#'
#' @source \url{http://www.ine.es}
"cod_muni_pjp_16"






data-raw/make_IGN_data.R
library(tidyverse)
library(stringr)
library(readxl)
library(readr)
library(here)
library(personal.pjp)
library(usethis)

#' (2018-02-04)-----------------------------------------------------
#- la web del [Instituto Geográfico Nacional](http://www.ign.es/web/ign/portal),
#- concretamente al [centro de descargas](http://centrodedescargas.cnig.es/CentroDescargas/index.jsp).
#- Una vez en el centro de descargas, hemos de ir a `Información geográfica de referencia`
#- y luego a `Lineas límite municipales`.
#- Trabajaré con los ficheros de "recintos".
#- Yo lo descargué el **27 de enero de 2018**.
#- Al descomprimirlo muchas carpetas: lineas y recintos para municipios, provincias y CC. AA.; además Canarias va aparte. Veámoslo:


#-------------------------- LINDES de CC.AA.
library(tidyverse)
library(sf)
library(here)
aa <- paste0(here(), "./data-raw/data_IGN/lineas_limite/recintos_autonomicas_inspire_peninbal_etrs89/recintos_autonomicas_inspire_peninbal_etrs89.shp")
ccaa_sf <- st_read(aa) #- 18 registros en Peninsula
aa <- paste0(here(), "./data-raw/data_IGN/lineas_limite/recintos_autonomicas_inspire_canarias_wgs84/recintos_autonomicas_inspire_canarias_wgs84.shp")
ccaa_can_sf <- st_read(aa)  #- 1 registo en Canarias
# cambio CRS de Canarias
ccaa_can_sf<- st_transform(ccaa_can_sf, 4258) #- convierto de epsg(SRID):4326   a  epsg(SRID): 4258
ccaa_todas <- rbind(ccaa_sf, ccaa_can_sf, deparse.level = 1) #- 19 registros
ccaa_todas <- ccaa_todas %>% mutate(INECodCCAA = str_extract(NATCODE, "^(....)"))   #- extraigo los 4 primeros caracteres de NATCODE
ccaa_todas <- ccaa_todas %>% mutate(INECodCCAA = str_extract(INECodCCAA, "(..)$"))   #- extraigo los 2 ultimos caracteres de INECodCCAA

# devtools::install_github("perezp44/pjppkgRdata01")  #- personal pkg
library(pjppkgRdata01) #- personal pkg
INE_prov <-  cod_provincias
INE_prov <- INE_prov %>% select(CODAUTO, CCAA) %>% distinct()
INE_prov <- INE_prov %>% rename(INECodCCAA = CODAUTO)
INE_prov <- INE_prov %>% rename(NombreCCAA = CCAA)
IGN_CCAA_17 <- left_join(ccaa_todas, INE_prov) %>% select(INECodCCAA, NATCODE, NAMEUNIT, NombreCCAA, everything())
IGN_CCAA_17s <- IGN_CCAA_17 %>% select(INECodCCAA, NATCODE, NombreCCAA, geometry)
# use_data(IGN_CCAA_17) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04
# use_data(IGN_CCAA_17s) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04



#-------------------------- LINDES de PROVINCIAS
#-------------------------- LINDES de PROVINCIAS

aa <- paste0(here(), "./data-raw/data_IGN/lineas_limite/recintos_provinciales_inspire_peninbal_etrs89/recintos_provinciales_inspire_peninbal_etrs89.shp")
prov_sf <- st_read(aa) #- 50 registros en Peninsula
aa <- paste0(here(), "./data-raw/data_IGN/lineas_limite/recintos_provinciales_inspire_canarias_wgs84/recintos_provinciales_inspire_canarias_wgs84.shp")
prov_can_sf <- st_read(aa)  #- 2 registos en Canarias
prov_can_sf<- st_transform(prov_can_sf, 4258) #- convierto de epsg(SRID):4326   a  epsg(SRID): 4258
#- para unirlos hay q usar rbind(..., deparse.level = 1)
prov_todas <- rbind(prov_sf, prov_can_sf, deparse.level = 1) #- 52 registros
prov_todas <- prov_todas %>% mutate(INECodProv = str_extract(NATCODE, "(.......)$"))   #- extraigo los 7 ultimos caracteres de NATCODE
prov_todas <- prov_todas %>% mutate(INECodProv = str_extract(INECodProv, "^(..)"))   #- extraigo los 2 primeros caracteres de INECodProv


INE_prov <-  cod_provincias
INE_prov <- INE_prov %>% rename(INECodProv = CPRO)
INE_prov <- INE_prov %>% rename(NombreProv = PROVINCIA)
INE_prov <- INE_prov %>% rename(INECodCCAA = CODAUTO)
INE_prov <- INE_prov %>% rename(NombreCCAA = CCAA)
IGN_prov_17 <- left_join(prov_todas, INE_prov) %>% select(INECodProv, NATCODE, NAMEUNIT, NombreProv, NombreCCAA, everything())
IGN_prov_17s <- IGN_prov_17 %>% select(INECodProv, NATCODE, NombreProv, INECodCCAA, NombreCCAA, geometry)

# use_data(IGN_prov_17) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04
# use_data(IGN_prov_17s) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04



#-------------------------- LINDES de MUNICIPIOS
#-------------------------- LINDES de MUNICIPIOS


aa <- paste0(here(), "./data-raw/data_IGN/lineas_limite/recintos_municipales_inspire_peninbal_etrs89/recintos_municipales_inspire_peninbal_etrs89.shp")
mun_sf <- st_read(aa) #- 50 registros en Peninsula
aa <- paste0(here(), "./data-raw/data_IGN/lineas_limite/recintos_municipales_inspire_canarias_wgs84/recintos_municipales_inspire_canarias_wgs84.shp")
mun_can_sf <- st_read(aa)  #- 2 registos en Canarias

mun_can_sf<- st_transform(mun_can_sf, 4258) #- convierto de epsg(SRID):4326   a  epsg(SRID): 4258
#- para unirlos hay q usar rbind(..., deparse.level = 1)
mun_todas <- rbind(mun_sf, mun_can_sf, deparse.level = 1) #- 8211 registros

mun_todas <- mun_todas %>% mutate(INECodMuni = str_extract(NATCODE, "(.....)$"))   #- extraigo los 5 ultimos caracteres de NATCODE

#- identifiqué las rows donde estaban las islitas alrededor de La Gomera
islitas_Gomera <- c(8156L, 8158L, 8177L, 8193L, 8207L, 8209L) #- están son las 6 rows que voy a poner q su 38 de INECodMuni pase a ser 66
mun_todas <- mun_todas %>% mutate(INECodMuni = str_extract(NATCODE, "(.....)$"))   #- extraigo los 5 ultimos caracteres de NATCODE
for (i in islitas_Gomera) {
  mun_todas$INECodMuni[i]  <- str_replace(mun_todas$INECodMuni[i] ,"38", "66")
}

INE_mun <-  cod_muni_pjp_17   #- 8124 municipios (1-enero-2017)
IGN_mun_17 <- left_join(mun_todas, INE_mun) %>% select(INECodMuni, NATCODE, NAMEUNIT, NombreMuni, NombreProv, NombreCCAA, everything())
IGN_mun_17s <- IGN_mun_17 %>% select(INECodMuni, NATCODE, NombreMuni, INECodProv, INECodCCAA, NombreProv, NombreCCAA, geometry)

# use_data(IGN_mun_17) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04
# use_data(IGN_mun_17s) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04





data-raw/make_INE_data.R
library(tidyverse)
library(stringr)
library(readxl)
library(readr)
library(here)
library(personal.pjp)
library(usethis)


#' (2018-02-04)-----------------------------------------------------

## Fichero INE con los códigos de las provincias y CCAA
cod_INE_prov <- read_excel(paste0(here(), "./data-raw/INE_data/cod_provincias.xlsx"))
cod_INE_prov <- cod_INE_prov %>% map(str_trim, side = "both") %>% as.tibble() #- quita caracteres al final y al ppio data.frame(lapply(eur_tot, trimws)) #
aa <- names_v_df_pjp(cod_INE_prov) #- ok, todo character
# devtools::use_data(cod_INE_prov) #- antes lo guardaba así (con devtools)
# use_data(cod_INE_prov) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04

## fichero del INE con los municipios a 1-enero-2017
cod_INE_muni_17 <- read_excel(paste0(here(), "./data-raw/INE_data/cod_municipios_2017.xlsx"), skip = 1, sheet = "dic17")
cod_INE_muni_17 <- cod_INE_muni_17 %>% map(str_trim, side = "both") %>% as.tibble() #- quita caracteres al final y al ppio data.frame(lapply(eur_tot, trimws)) #
aa <- names_v_df_pjp(cod_INE_muni_17) #- ok, todo character
# use_data(cod_INE_muni_17) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04


## fichero del INE con los municipios a 1-enero-2016
cod_INE_muni_16 <- read_excel(paste0(here(), "./data-raw/INE_data/cod_municipios_2017.xlsx"), skip = 1, sheet = "dic16")
cod_INE_muni_16 <- cod_INE_muni_16 %>% map(str_trim, side = "both") %>% as.tibble() #- quita caracteres al final y al ppio data.frame(lapply(eur_tot, trimws)) #
aa <- names_v_df_pjp(cod_INE_muni_16) #- ok, todo character
# use_data(cod_INE_muni_16) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04



## Voy a fusionar cod_municipios_17 con cod_provincias
## Primero creo el codigo mas habitual en INE: INECodMuni
cod_INE_muni_17 <- cod_INE_muni_17 %>% mutate(INECodMuni = paste0(CPRO, CMUN)) #- Hay 8124 municipios.
cod_INE_prov <- cod_INE_prov %>% select(-CODAUTO) #- la quito xq sino se duplicaria
#- el fichero lo llamo cod_INE_muni_pjp
cod_muni_pjp_17 <-  left_join(cod_INE_muni_17, cod_INE_prov, by = "CPRO")
aa <- names_v_df_pjp(cod_muni_pjp_17) #- ok, todo character
#- renombro las primeras columnas para hacerlas compatibles con "municipios_CNIG"
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% rename(NombreMuni = NOMBRE)
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% rename(NombreProv = PROVINCIA)
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% rename(NombreCCAA = CCAA)
#- cambio nombre de las series de cogigos, la de los municipios ya esta hecha (INECodMuni)
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% rename(INECodProv = CPRO)
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% rename(INECodCCAA = CODAUTO)
#- Creo version numerica de los codigos INE de CCAA, prov y municipios
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% mutate(INENumMuni = as.integer(INECodMuni))
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% mutate(INENumProv = as.integer(INECodProv))
cod_muni_pjp_17 <- cod_muni_pjp_17 %>% mutate(INENumCCAA = as.integer(INECodCCAA))
aa <- names_v_df_pjp(cod_muni_pjp_17) #- ok, todo character y 3 integers
# use_data(cod_muni_pjp_17) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04





## Voy a fusionar cod_municipios_16 con cod_provincias
## Primero creo el codigo mas habitual en INE: INECodMuni
cod_INE_muni_16 <- cod_INE_muni_16 %>% mutate(INECodMuni = paste0(CPRO, CMUN)) #- Hay 8125 municipios.
cod_INE_prov <- cod_INE_prov %>% select(-CODAUTO) #- la quito xq sino se duplicaria
#- el fichero lo llamo cod_INE_muni_pjp
cod_muni_pjp_16 <-  left_join(cod_INE_muni_16, cod_INE_prov, by = "CPRO")
aa <- names_v_df_pjp(cod_muni_pjp_16) #- ok, todo character
#- renombro las primeras columnas para hacerlas compatibles con "municipios_CNIG"
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% rename(NombreMuni = NOMBRE)
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% rename(NombreProv = PROVINCIA)
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% rename(NombreCCAA = CCAA)
#- cambio nombre de las series de cogigos, la de los municipios ya esta hecha (INECodMuni)
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% rename(INECodProv = CPRO)
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% rename(INECodCCAA = CODAUTO)
#- Creo version numerica de los codigos INE de CCAA, prov y municipios
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% mutate(INENumMuni = as.integer(INECodMuni))
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% mutate(INENumProv = as.integer(INECodProv))
cod_muni_pjp_16 <- cod_muni_pjp_16 %>% mutate(INENumCCAA = as.integer(INECodCCAA))
aa <- names_v_df_pjp(cod_muni_pjp_16) #- ok, todo character y 3 integers
# use_data(cod_muni_pjp_16) #- lo guarde con esta linea que usa el pkg usethis el 2018-02-04





data-raw/scripts_old-goo/_script_add_data.R
#- PARA AÑADIR un nuevo df u objeto:
#- 1) Cargas el objeto en memoria de R con el archivo ./data-raw/make-datasets.R
#- 2) Para q vaya a memoria del pkg has de: devtools::use_data(CNIG_CCAA_sf, overwrite = TRUE) #
#- 3) Añadir la documentacion co roxygen en ./R/data.R
#- 4) Finalmente has de actualizar con: devtools::document()
#- 5) Llevarlo a Github

library(readxl)
#dir.create("./data")


# cod_provincias <- read_excel("./data-raw/cod_provincias.xlsx")
# devtools::use_data(cod_provincias)
# save(cod_provincias, file="./data/cod_provincias.RData") #- guardamos my_data en formato .RData o .rda


#- AQUI
#- acuerdate de documentar cada nuevo fichero de datos q añadas en el fichero:
#- ./R./data.R


#- Para q se modifique el NAMESPACE
devtools::document()


#- VIÑETAS
#devtools::use_vignette("my_vignette")


#----- Instalarlo dsde Local
# install.packages("C:/Users/perezp/Desktop/a_GIT_2016a/pjppkgRdata01", repos = NULL, type="source")
# library(mypkgfordata)
# aa <- data("cod_provincias")

#--- instalarlo desde Github
#devtools::install_github("perezp44/pjppkgRdata01")

#- para ver los datos q tengo

library(tidyverse)
library(pjppkgRdata01)

data(cod_provincias)

ls("package:pjppkgRdata01", all = TRUE) #- ves lo que hay en mypkgDataforblog
aa <- ls("package:pjppkgRdata01", all = TRUE) %>% as.data.frame()#- ves lo que hay en mypkgDataforblog


help(package = pjppkgRdata01)
??pjppkgRdata01

vignette(package = "pjppkgRdata01")

bb <- cod_muni_pjp_17


#- Github

cd c:/Users/perezp/Desktop/a_GIT_2016a/mypkgDataforblog

git add -A    # stages all files
git add .
git commit --all --message "Creando el REPO"
git remote add origin https://github.com/perezp44/mypkgfordata.git

git push -u origin master



git add -A         # stages all files
git add .          # stages new and modified, without deleted
git add -u         # stages modified and deleted, without new
git add file.R     # stages just the 'file.R' file




R/IGN_data_lindes.R
#------------------------------ This file documents IGN datasets para los LINDES

#' LINDES del IGN para CC.AA (SPAIN)
#'
#' Poligonos/geometrias/shapes de las provincias  CC.AA. (SPAIN). Sacados del IGN
#' 19 CC.AA (18 = Ceuta y 19 = Melilla).
#'
#' @source \url{http://centrodedescargas.cnig.es/CentroDescargas/index.jsp}
#'
#' @format A sf-data frame with 19 rows and 11 variables:
#' \itemize{
#'   \item INECodCCAA: Codigo INE de la C.A (Galicia = 12)
#'   \item NATCODE: Codigo del IGN (Galicia = 34120000000)
#'   \item NAMEUNIT: Nombre de la CC.AA (IGN)
#'   \item NombreCCAA: Nombre de la CC.AA (INE)
#'   \item INSPIREID: IGN code (Galicia = ES.IGN.SIGLIM34120000000)
#'   \item COUNTRY: Country code (ES = Spain)
#'   \item NATLEVEL: http://inspire.ec.europa.eu/codelist/AdministrativeHierarchyLevel/2ndOrder
#'   \item CODNUT1: (Galicia = ES1)
#'   \item CODNUT2: (Galicia = ES1)
#'   \item CODNUT3: NA (no hay datos)
#'   \item geometry: la geometría
#'   }
#'
#' @examples
#' \dontrun{
#'  df <- IGN_CCAA_17
#' }
#'
"IGN_CCAA_17"


#' LINDES del IGN para CC.AA (SPAIN) (version short)
#'
#' Poligonos/geometrias/shapes de las provincias  CC.AA. (SPAIN). Sacados del IGN
#' 19 CC.AA (18 = Ceuta y 19 = Melilla). Solo 4 variables
#'
#' @source \url{http://centrodedescargas.cnig.es/CentroDescargas/index.jsp}
#'
#' @format A sf-data frame with 19 rows and 4 variables:
#' \itemize{
#'   \item INECodCCAA: Codigo INE de la C.A (Galicia = 12)
#'   \item NATCODE: Codigo del IGN (Galicia = 34120000000)
#'   \item NombreCCAA: Nombre de la CC.AA (INE)
#'   \item geometry: la geometria
#'   }
#'
#' @examples
#' \dontrun{
#'  df <- IGN_CCAA_17s
#' }
#'
"IGN_CCAA_17s"



#' LINDES del IGN para PROVINCIAS (SPAIN)
#'
#' Poligonos/geometrias/shapes de las provincias  CC.AA. (SPAIN). Sacados del IGN
#' Codigos y nombres oficiales de las provincias y CCAA (SPAIN). Sacados del INE.
#' 52 provincias (51 = Ceuta y 52 = Melilla).
#'
#' @source \url{http://centrodedescargas.cnig.es/CentroDescargas/index.jsp}
#'
#' @format A sf-data frame with 52 rows and 13 variables:
#' \itemize{
#'   \item INECodProv: Codigo provincia (INE) (Teruel = 44)
#'   \item NATCODE: Code IGN (Teruel = 34024400000)
#'   \item NAMEUNIT: Nombre de la provincia (IGN)
#'   \item NombreProv: Nombre de la provincia (INE)
#'   \item NombreCCAA: Nombre C.A (INE)
#'   \item INSPIREID: Teruel = ES.IGN.SIGLIM34024400000
#'   \item COUNTRY: ES = Spain
#'   \item NATLEVEL: http://inspire.ec.europa.eu/codelist/AdministrativeHierarchyLevel/3rdOrder
#'   \item CODNUT1: (Teruel = ES2)
#'   \item CODNUT2: (Teruel = ES24)
#'   \item CODNUT3: NA
#'   \item INECodCCAA: Codigo de la C.A (INE)
#'   \item geometry: la geometria
#' }
#'
#' @examples
#' \dontrun{
#'  df <- IGN_prov_17
#' }
#'
"IGN_prov_17"


#' LINDES del IGN para PROVINCIAS (SPAIN) (Short version)
#'
#' Poligonos/geometrias/shapes de las provincias  CC.AA. (SPAIN). Sacados del IGN
#' Codigos y nombres oficiales de las provincias y CCAA (SPAIN). Sacados del INE.
#' 52 provincias (51 = Ceuta y 52 = Melilla).
#'
#' @source \url{http://centrodedescargas.cnig.es/CentroDescargas/index.jsp}
#'
#' @format A sf-data frame with 52 rows and 6 variables:
#' \itemize{
#'   \item INECodProv: Código provincia (INE) (Teruel = 44)
#'   \item NATCODE: Code IGN (Teruel = 34024400000)
#'   \item NombreProv: Nombre de la provincia (INE)
#'   \item NombreCCAA: Nombre C.A (INE)
#'   \item INECodCCAA: Código de la C.A (INE)
#'   \item geometry: la geometria
#' }
#'
#' @examples
#' \dontrun{
#'  df <- IGN_prov_17
#' }
#'
"IGN_prov_17s"




#' LINDES del IGN para Municipios (SPAIN)
#'
#' Poligonos/geometrias/shapes de los municipios  CC.AA. (SPAIN). Sacados del IGN
#' Codigos y nombres oficiales de los municipios (SPAIN). Sacados del INE.
#' 8.124 municipios + 81 condominio + 6 islitas alrededor de La Gomera = 8.211 rows
#'
#' @source \url{http://centrodedescargas.cnig.es/CentroDescargas/index.jsp}
#'
#' @format A data frame with 8211 rows and 20 variables:
#' \itemize{
#'   \item INECodMuni: Codigo INE municipio (5 digitos (Pancrudo = 44177))
#'   \item NATCODE: Codigo IGN (Pancrudo = 34024444177)
#'   \item NAMEUNIT: Nombre municipio (IGN)
#'   \item NombreMuni: Nombre municipio (INE)
#'   \item NombreProv: Nombre provincia (INE)
#'   \item NombreCCAA: Nombre de la C.A (INE)
#'   \item INSPIREID: (Pancrudo = ES.IGN.SIGLIM34024444177)
#'   \item COUNTRY: (ES = Spain)
#'   \item NATLEVEL: http://inspire.ec.europa.eu/codelist/AdministrativeHierarchyLevel/4thOrder
#'   \item CODNUT1: (Pancrudo = ES2)
#'   \item CODNUT2: (Pancrudo = ES24)
#'   \item CODNUT3: (Pancrudo = ES242)
#'   \item INECodCCAA: Codigo CCAA(INE) (Pancrudo = 02)
#'   \item INECodProv: Codigo provincial(INE)(Pancrudo = 44)
#'   \item CMUN: Código municipal(INE)(Pancrudo = 177)
#'   \item DC: Digito control (INE) (pancrudo = 4)
#'   \item INENumMuni: Código municipal(INE) (numeric)(Pancrudo = 177)
#'   \item INENumProv: Codigo provincial(INE) (numeric)(Pancrudo = 44)
#'   \item INENumCCAA: Codigo CCAA(INE) (numeric)(Pancrudo = 02)
#'   \item geometry: la geometria
#' }
#'
#' @examples
#' \dontrun{
#'  df <- IGN_mun_17s
#' }
#'
"IGN_mun_17"





#' LINDES del IGN para Municipios (SPAIN) (short version)
#'
#' Poligonos/geometrias/shapes de los municipios  CC.AA. (SPAIN). Sacados del IGN
#' Codigos y nombres oficiales de los municipios (SPAIN). Sacados del INE.
#' 8.124 municipios + 81 condominio + 6 islitas alrededor de La Gomera = 8.211 rows
#'
#' @source \url{http://centrodedescargas.cnig.es/CentroDescargas/index.jsp}
#'
#' @format A data frame with 8211 rows and 7 variables:
#' \itemize{
#'   \item INECodMuni: Codigo INE municipio (5 digitos (Pancrudo = 44177))
#'   \item NATCODE: Codigo IGN (Pancrudo = 34024444177)
#'   \item NombreMuni: Nombre municipio (INE)
#'   \item INECodProv: Codigo provincial(INE)(Pancrudo = 44)
#'   \item INECodCCAA: Codigo CCAA(INE) (Pancrudo = 02)
#'   \item NombreProv: Nombre provincia (INE)
#'   \item NombreCCAA: Nombre de la C.A (INE)
#'   \item geometry: la geometria
#' }
#'
#' @examples
#' \dontrun{
#'  df <- IGN_mun_17s
#' }
#'
"IGN_mun_17s"





